### PR TITLE
Code refactor and simplification

### PR DESCRIPTION
**Pull request by zepiaf**
Tried to simplify and refactor the code

Event is now a class
Period is now a class

Both are accessed as objects (ie event.date) and can be dumped as dict (event.to_dict())

### CODE REVIEW

**Comment by zepiaf** [comment]
@mdallaire @clauderobi and anyone else :)

Please review, so we can get the code more readable and flexible

File: winter_credit/event.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+class Event:
+    """This class describe an event object"""
+
+    def __init__(self, config, date, start, end):
+        self.config = config
+        self.date_dt = date
+        self.start_dt = start
+        self.end_dt = end
+        self.date = date.strftime('%Y-%m-%d')
```
Comment (mdallaire) [inline winter_credit/event.py:11]: We should probably provide a config setting for date format like we do for datetime format.

**Comment by mdallaire** [comment]
Looks good to me!

File: winter_credit/event.py (hunk 1 above)
Comment (zepiaf) [reply winter_credit/event.py:11]: True
Let's add this to the todo list and add later

**Pull request merged**